flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/ResourceManager.java

```diff
@@ -874,6 +894,13 @@ public void handleError(final Exception exception) {
 	 */
 	public abstract boolean stopWorker(ResourceID resourceID);
 
+	/**
+	 * Cancel the allocation of a resource. If the resource allocation has not fulfilled, should cancel it.
+	 *
+	 * @param resourceProfile The resource description of the previous allocation
+	 */
+	public abstract void cancelNewWorker(ResourceProfile resourceProfile);
```

Comment:
See my comment in the `SlotManager`. I think we don't need this call here. Rather, I would like the SlotManager to decide upon registration of a new resource whether it needs it or not and release the resource via calling `ResourceManagerActions#releaseResource`.

Reply:
I comment it is slot manager.